### PR TITLE
Add nonProxyHosts/targetProxyHosts support to OAuth token endpoint proxy configuration

### PR DESCRIPTION
**Pull request by Tharsanan1**
## Summary
Adds `nonProxyHosts` and `targetProxyHosts` support to the Synapse-level OAuth token endpoint proxy configuration. This was lost when APIM migrated from class-mediator-based OAuth to the Synapse endpoint implementation.

**Issue:** https://github.com/wso2/api-manager/issues/5003

## Root Cause
The `ProxyConfigs` class and `OAuthUtils.getSecureClient()` did not support `nonProxyHosts` or `targetProxyHosts` fields. The HTTP client always used `DefaultProxyRoutePlanner` which routes ALL traffic through the proxy unconditionally.

### ProxyConfigs.java
- Added `nonProxyHosts` and `targetProxyHosts` String fields with getters and setters

### AuthConstants.java
- Added 4 constants: `PROXY_NON_PROXY_HOSTS`, `PROXY_TARGET_PROXY_HOSTS`, `OAUTH_GLOBAL_PROXY_NON_PROXY_HOSTS`, `OAUTH_GLOBAL_PROXY_TARGET_PROXY_HOSTS`

### OAuthUtils.java
- `getProxyConfigs()`: Reads new fields from endpoint XML and global synapse.properties
- `createOMProxyConfigs()`: Serializes new fields
- `getSecureClient()`: Replaced `DefaultProxyRoutePlanner` with `NonProxyAwareProxyRoutePlanner`
- Added `NonProxyAwareProxyRoutePlanner` inner class that evaluates glob patterns for proxy bypass/restrict

### New Test Files
- `ProxyConfigsTest.java` — 5 tests for getter/setter
- `NonProxyAwareProxyRoutePlannerTest.java` — 16 tests covering: exact match, wildcard, pipe-separated, null/empty, targetProxyHosts restrict, precedence, edge cases

## Verification
- All 21 unit tests pass
- End-to-end test with real HTTP proxy and echo backend confirmed:
  - Requests matching `targetProxyHosts` route through proxy
  - Requests NOT matching bypass proxy
  - `nonProxyHosts` takes precedence over `targetProxyHosts`

### CODE REVIEW

**Review by npamudika — DISMISSED** [review]

File: modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/OAuthUtils.java [hunk 1]
```diff
@@ -948,4 +968,84 @@ public static String resolveProxyPassword(ProxyConfigs proxyConfigs, MessageCont
...
+                }
+                return null;
+            }
+
+            return super.determineProxy(target, request, context);
+        }
+
+        private boolean matchesPattern(String hostname, String pattern) {
+            String regex = pattern.replace(".", "\\.").replace("*", ".*");
+            return hostname.matches(regex);
```
[the remark this answers is not in the document]
Comment (Tharsanan1) [reply modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/OAuthUtils.java]: I have fixed this please review again and approve

File: modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/OAuthUtils.java [hunk 2]
```diff
@@ -948,4 +970,109 @@ public static String resolveProxyPassword(ProxyConfigs proxyConfigs, MessageCont
...
+                return null;
+            }
+
+            return super.determineProxy(target, request, context);
+        }
+
+        private boolean matchesPattern(String hostname, String pattern) {
+            String normalizedHost = hostname.toLowerCase(Locale.ROOT);
+            String normalizedPattern = pattern.toLowerCase(Locale.ROOT);
+            StringBuilder regex = new StringBuilder(normalizedPattern.length() + 8);
```
Comment (GDLMadushanka) [inline modules/core/src/main/java/org/apache/synapse/endpoints/auth/oauth/OAuthUtils.java]: Shall we add a constant with an explanation for this buffer 8 ?

**Review by GDLMadushanka — APPROVED** [review]
